game-core/src/main/java/games/strategy/triplea/util/MovableUnitsFilter.java

```diff
@@ -223,15 +224,10 @@ private MoveValidationResultWithDependents validateMoveWithDependents(
       final List<Unit> best, final Map<Unit, Collection<Unit>> dependentUnits) {
     final MustMoveWithDetails mustMoveWithDetails =
         MoveValidator.getMustMoveWith(route.getStart(), dependentUnits, player);
-    final List<Unit> bestWithDependents = new ArrayList<>(best);
+    final var bestWithDependents = new HashSet<>(best);
     for (final Unit u : best) {
-      final Collection<Unit> mustMoveWith = mustMoveWithDetails.getMustMoveWithForUnit(u);
-      for (final Unit m : mustMoveWith) {
-        if (!bestWithDependents.contains(m)) {
-          bestWithDependents.addAll(mustMoveWith);
-        }
-      }
+      bestWithDependents.addAll(mustMoveWithDetails.getMustMoveWithForUnit(u));
     }
-    return bestWithDependents;
+    return new ArrayList<>(bestWithDependents);
```

Comment:
I briefly checked the callers of this method, and I'm almost sure you could return a Set here directly.
Of course this would require some refacoring in other places to accept `Collection` instead of `List`, but that should really be it. Avoids the need to create a list-copy of the set

Reply:
It might be possible, but quickly checking it would make this PR much bigger - since the type ends up making it to unitsWithDependents on FilterOperationResult and thus part of the API.

Let me look at doing that in a separate PR after this one instead.

Reply:
Yeah of course, I didn't want you to do this as part of this PR.
So that's fine